Engines/Wine/Plugins/Windows version/script.js

```diff
@@ -1,72 +1,69 @@
-const Wine = include("engines.wine.engine.object");
-
-include("engines.wine.plugins.regedit");
+const Regedit = include("engines.wine.plugins.regedit");
 
 /**
- * default windows version
- * @param {string} version (win7, vista, win2003, winxp, win2k, winnt, winme, win98, win95, win31)
- * @param {string} [servicePack] e.g. sp3
- * @returns {string|Wine} get: Windows version, set: Wine object
+ * Plugin to set the windows version
  */
-Wine.prototype.windowsVersion = function (version, servicePack) {
-    // get
-    if (arguments.length == 0) {
-        return this.regedit().fetchValue(["HKEY_CURRENT_USER", "Software", "Wine", "Version"]);
+module.default = class WindowsVersion {
+    constructor(wine) {
+        this.wine = wine;
+        this.applications = {};
     }
 
-    // set
-    var regeditFileContent;
-    if (version == null) {
-        regeditFileContent =
-            "REGEDIT4\n" +
-            "\n" +
-            "[HKEY_CURRENT_USER\\Software\\Wine]\n" +
-            "\"Version\"=-\n";
-    } else {
-        regeditFileContent =
-            "REGEDIT4\n" +
-            "\n" +
-            "[HKEY_CURRENT_USER\\Software\\Wine]\n" +
-            "\"Version\"=\"" + version + "\"\n";
+    getWindowsVersion() {
+        return new Regedit(this.wine).fetchValue(["HKEY_CURRENT_USER", "Software", "Wine", "Version"]);
+    }
 
-        if (servicePack) {
-            var servicePackNumber = servicePack.replace("sp", "");
-            regeditFileContent += "[HKEY_LOCAL_MACHINE\\Software\\Microsoft\\Windows NT\\CurrentVersion]\n";
-            regeditFileContent += "\"CSDVersion\"=\"Service Pack " + servicePackNumber + "\"\n";
-            regeditFileContent += "[HKEY_LOCAL_MACHINE\\System\\CurrentControlSet\\Control\\Windows]\n";
-            regeditFileContent += "\"CSDVersion\"=dword:00000" + servicePackNumber + "00\n";
-        }
+    /**
+     * Specifies the default windows version
+     *
+     * @param {string} version win7, vista, win2003, winxp, win2k, winnt, winme, win98, win95, win31
+     * @param {string} servicePack e.g. sp3
+     * @returns {WindowsVersion} This
+     */
+    withWindowsVersion(version, servicePack) {
+        this.version = version;
+        this.servicePack = servicePack;
+
+        return this;
     }
 
-    this.regedit().patch(regeditFileContent);
-    return this;
-};
+    /**
+     * Specifies the windows version for a given application
+     * @param {string} application The application executable
+     * @param {string} os The windows version
+     * @returns {WindowsVersion} This
+     */
+    withApplicationWindowsVersion(application, os) {
+        this.applications[application] = os;
 
-var SetOsForApplication = function () {
-    var that = this;
-    that._regeditFileContent =
-        "REGEDIT4\n" +
-        "\n";
+        return this;
+    }
 
-    that.wine = function (wine) {
-        that._wine = wine;
-        return that;
-    };
+    go() {
```

Comment:
There are some plugins which were previously two separate `Wine` functions. One of these is the `WindowsVersion` plugin which previously consisted of:
- `windowsVersion` and
- `setOsForApplication`

I've tried to unify all such cases into a single plugin. This may lead to some strange behavior compared to before because the new behavior may differ if a plugin is used twice, once to set the global windows version and then a second time to set a local application-specific windows version